grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -1486,4 +1486,87 @@ class QueryService implements Status {
         }
         results
     }
+
+
+    PaginatedResponse findDataUseRestrictions(PaginationParams pagination) {
+        String query = 'select count(id) from data_use_restriction '
+        String orderColumn = getRestrictionOrderColumn(pagination.orderColumn)
+        if (pagination.searchValue) {
+            query = query + 'where consent_group_key LIKE :term OR vault_export_date LIKE :term '
+        }
+        SQLQuery sqlQuery = getSessionFactory().currentSession.createSQLQuery(query)
+        if (pagination.searchValue) sqlQuery.setString('term', pagination.getLikeTerm())
+        Long totalRows = sqlQuery.uniqueResult()
+        if (orderColumn) {
+            query = query + " order by " + orderColumn + pagination.sortDirection
+        }
+        sqlQuery = getSessionFactory().currentSession.createSQLQuery(query.replace('count(id)', '*'))
+        List<DataUseRestriction> results = sqlQuery.with {
+            if (pagination.searchValue) setString('term', pagination.getLikeTerm())
+            addEntity(DataUseRestriction)
+            setFirstResult(pagination.start)
+            setMaxResults(pagination.length)
+            list()
+        }
+        new PaginatedResponse(
+                draw: pagination.draw,
+                recordsTotal: totalRows,
+                recordsFiltered: totalRows,
+                data: results,
+                error: ""
+        )
+    }
+
+    Collection findCollectionLinks() {
+        String query = 'select * from consent_collection_link where deleted = 0 '
+        SQLQuery sqlQuery = getSessionFactory().currentSession.createSQLQuery(query)
+        List<ConsentCollectionLink> links = sqlQuery.with {
+            addEntity(ConsentCollectionLink)
+            list()
+        }
+        Map<String, SampleCollection> collectionMap = getCollectionIdMap(links)
+        Collection<DataUseRestriction> durs = findDataUseRestrictionByConsentGroupKeyInList(links.collect {
+            it.consentKey
+        })
+        links.each { link ->
```

Comment:
There is some kind of weirdness with respect to how SQLQuery.with types the results it returns. It comes back as an untyped collection, even if the left hand side of the assignment declares the type. If you look around in this class, you'll see that we have to cast the results. I wish there were a better way to do this, but I don't know what that is. In the meantime, this will ensure that links is what you think it is here:
```suggestion
        (links as Collection<ConsentCollectionLink>).each { link ->
```